infra.bs

```diff
@@ -569,6 +569,29 @@ the <a>string</a> with their corresponding <a>code point</a> in <a>ASCII upper a
 <a>ASCII lowercase</a> of <var>B</var>.
 <!-- TODO: define string equals? -->
 
+<p>To <dfn export>ASCII encode</dfn> a <a>string</a> <var>input</var>, run these steps:</p>
+
+<ol>
+ <li><p><a>Assert</a>: <var>input</var> is an <a>ASCII string</a>.
+
+ <p class=note>Note: This precondition ensures that <a>isomorphic encode</a> and
+ <a>utf-8 encode</a> return the same <a>byte sequence</a> for this input.
```

Comment:
Nit: looks like the source material uppercases UTF-8. Here and below.

Reply:
Whoops, thanks.